storage/innobase/include/dict0dd.h

```diff
@@ -1352,5 +1353,15 @@ the dictionary.
 @return true if it does. */
 bool dd_is_table_in_encrypted_tablespace(const dict_table_t *table);
 
+/** Check if InnoDB must access MySQL data dictionary through the server layer
+APIs. If InnoDB is the data dictionary storage engine, it may access the data
+dictionary through lower level direct InnoDB interface. But if it is not the
+DDSE, the tables will be in another storage engine, thus the server layer
+interface must be used.
+@return whether server layer interface must be used to access the DD */
+[[nodiscard]] inline bool dd_access_through_server() noexcept {
+  return (default_dd_storage_engine != DEFAULT_DD_INNODB);
```

Comment:
how about add a new variable? with that variable enabled(it will always access through server), maybe we can reuse more innodb MTRs?

Reply:
Possible, but I view adding new system variables as last resort, because MySQL already has the problem of too many sysvars

Reply:
I have also considered making the new method mandatory for InnoDB DDSE too, but that might be quite risky

Reply:
As discussed elsewhere, let's keep as is.